std/container/binaryheap.d

```diff
@@ -282,7 +282,7 @@ and $(D length == capacity), throws an exception.
             import std.traits : isDynamicArray;
             static if (isDynamicArray!Store)
             {
-                if (_store.length == 0)
+                if (_store.length < 6)
```

Comment:
This logic needs to be reorganized. this is why the auto tester is failing. Some algorithms expect the binary heap to stay fixed size, and leave the array in place instead of reallocating. Here, we are reallocating even if we don't need to.

So I think it should be:
```D
if(length == _store.length)
    _store.length = (length < 6 ? 8 : length * 3 / 2)

```

This should cover the 0 and 1 cases exactly as we need it to, but only append length if we are out of room.